### ISSUE TITLE
BaseUrl is wrong when Resource is absolute (with protocol)

### ISSUE THREAD

**Issue by datvm** [restsharp/RestSharp#1035]
In this issue: https://github.com/restsharp/RestSharp/issues/619 you are asking a Use case for `BaseUrl` should not null. I actually have a problem with IdentityServer (actually just a normal OAuth2 server):

`this.RestClient = new RestClient(this.Config.Authority);`

For example, the Url should be `https://www.example1.com/`

Then I perform a Discovery:

`var request = new RestRequest(".well-known/openid-configuration", Method.GET);`

And I can get the Endpoints. However, the response will be:

`"authorization_endpoint": "https://www.example1.com/connect/authorize",`

Now, I cannot use that RestClient, because:

`var request = new RestRequest(this.AuthorizeEndpoint, Method.GET);`

will returns this Uri:

`https://www.example1.com/https://www.example1.com/connect/authorize?`

The desirable should be, when a RestRequest resource is absolute, it should not be appended at all. I tried with Uri and this one is correct, why is it not the case for RestRequest?

```
            var a = new Uri("https://www.example1.com/");
            var b = new Uri(a, "https://www.example1.com/connect/authorize");
            var c = b.AbsoluteUri; // c is "https://www.example1.com/connect/authorize"
```

**Comment by alexeyzimarev** [comment]
@datvm I merged the PR and the 106.1 package contains your fix. Could you please test if the issue is resolved and close the issue?

**Comment by datvm** [comment]
Hi I have checked the 106.1. It's fixed. Thank you very much!
